### ISSUE TITLE
UX bug: Mautic uses different ways to filter URLs across the application: campaign decisions and points

### ISSUE THREAD

**Issue by adiux** [mautic/mautic#12087]
### Mautic Version

4.4.x series

### PHP version

7.4

### What browsers are you seeing the problem on?

Not relevant

### What happened?

Depends on the place in the app Mautic uses a different logic to filter URLs. 

Some use a [shell wildcard pattern](https://www.php.net/manual/en/function.fnmatch.php) like `*test*`
- Points
- Visited a page

Some use [regular expressions](https://www.php.net/manual/en/function.preg-match.php) like `test`
- Clicks a URL in an email

So if I want to search for a string 'test' inside a URL I have to use `*test*` or `test` depending what feature of Mautic I'm using. 

This should be consolidated into one. 

IMO it should be all regular expressions. This is what people are used from e.g. Google Analytics, and it does not need any wildcards for a standard string in string search case. See the example above.

We can use the `UrlMatcher::hasMatch` methode.

I can do a PR if we agree on the method to be used everywhere: shell or regex?

### How can we reproduce this issue?

Step 1: Check for a string in a URL in the campaign decision: Clicks link with the filter `*test*`
Step 2: Check for a string in a URL in the campaign decision: Visited URL `*test*`
Step 3: Visited a URL will evaluate to true, Clicked a link will evaluate to false

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by escopecz** [comment]
Definitely a good idea to unify it. Please make the PR. I agree that we should use the same methods as other tools like GA. Can you clarify where do you search for the URLs? In reports?

**Comment by adiux** [comment]
@escopecz Thanks John. I became aware of it in Campaign decisions and Points. If I remember correctly, then the Campaign decisions "Clicks email" and "Visits page" also behave different.

**Comment by escopecz** [comment]
We should get it in M5 and document what could break with this change so users will double-check their campaigns. Or do you think that a migration can take care of existing campaigns and points?

**Comment by adiux** [comment]
I guess a migration could take care of it partially. E.g. we can just remove the stars: *test* and it will work as before. But it will be hard to replace all and very complex cases. Maybe better to have it in the release notes with some examples?

**Comment by escopecz** [comment]
Yep. If the migration takes care of 80% of cases then it's a big help.

**Comment by adiux** [comment]
We are working on this, a fix should be available soon.
